frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MovieListFragment.java

```diff
@@ -0,0 +1,227 @@
+        bundle.putInt("position", position);
+        bundle.putString("type", type);
+        toolbar.setTitle(movieListAdapter.getItem(position).getTitle());
+        toolbar.setSubtitle(null);
+        moviePageFragment.setArguments(bundle);
+        fragmentTransaction.replace(R.id.fl_content,moviePageFragment);
+        fragmentTransaction.addToBackStack(null);
+        fragmentTransaction.commit();
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        EventBus.getDefault().unregister(this);
+    }
+
+    @Subscribe
+    public void onEvent(AddMovieToMovieListEvent event){
+        movieListAdapter.addMovieItemToAdapter(event.movie, event.position);
+        Log.i(TAG, "New movie is added to local list");
+    }
+
+    @Subscribe
+    public void onEvent(DeleteMovieFromMovieListEvent event){
+        movieListAdapter.removeMovieItemToAdapter(event.position);
+        Log.i(TAG, "A movie is removed from local list");
+    }
+
+    @Subscribe
+    public void onEvent(UpdateMovieListEvent event){
+        movieListAdapter.updateMovieItemToAdapter(event.movie, event.position);
+        Log.i(TAG, "A movie is updated to local list");
+    }
+
+    //Async thread to handle search request
+    private class SearchMoreTask extends AsyncTask<Void, Integer, Void> {
+        private final int SUCCESSFUL = 0;
+        private final int ERROR = 1;
+        private final int NO_RESULT = 2;
+        private final int NO_INTERNET = 3;
+        private final int NO_MORE_RESULT = 4;
+        int status;
+
+        @Override
+        protected Void doInBackground(Void... params) {
+            try {
+                searchRequestHttpBuilder = RequestHttpBuilderSingleton.getInstance();
+                OkHttpClient client = searchRequestHttpBuilder.getClient();
+                Request request = searchRequestHttpBuilder.getNextSearchRequest();
+                Response response = client.newCall(request).execute();
+                if (!response.isSuccessful()) {
+                    Log.i(TAG, "connect failed");
+                    status = ERROR;
+                    throw new IOException("Unexpected code" + response);
+                }
+                //Convert json to Arraylist<Movie>
+                movieListWithCount = gson.fromJson(response.body().charStream(), new TypeToken<MovieWithCount>() {}.getType());
+                resultCount = movieListWithCount.getCount();
+                searchMovieList = movieListWithCount.getMovies();
+
+                if (resultCount == 0) {
+                    status = NO_RESULT;
+                }
+                else if(searchMovieList.size() < searchRequestHttpBuilder.getLimit()){
+                    status = NO_MORE_RESULT;
+                }
+                else {
+                    status = SUCCESSFUL;
+                }
+                return null;
+            } catch (IOException e) {
+                status = NO_INTERNET;
```

Comment:
This IOException may also occur when our server is down thus it does not necessarily mean the problem lies within the user's control.